shared/chat/conversation/input-area/normal/platform-input.desktop.js

```diff
@@ -201,148 +180,157 @@ class PlatformInput extends React.Component<PlatformInputProps & Kb.OverlayParen
-                ? Styles.globalColors.black_75
-                : Styles.globalColors.black_20,
-            },
-          ])}
-        >
-          {!this.props.isEditing && (
-            <HoverBox
-              className={Styles.classNames({expanded: this.props.showingMenu})}
-              onClick={this._toggleShowingMenu}
-              ref={this.props.setAttachmentRef}
-              style={Styles.collapseStyles([
-                styles.explodingIconContainer,
-                !!this.props.explodingModeSeconds && {
-                  backgroundColor: Styles.globalColors.black_75,
-                },
-              ])}
-            >
-              {this.props.explodingModeSeconds ? (
-                <Kb.Text type="BodyTinyBold" style={styles.time}>
-                  {formatDurationShort(this.props.explodingModeSeconds * 1000)}
-                </Kb.Text>
-              ) : (
-                <Kb.Icon
-                  className="timer"
-                  onClick={this._toggleShowingMenu}
-                  style={Kb.iconCastPlatformStyles(styles.timerIcon)}
-                  type="iconfont-timer"
-                />
-              )}
-            </HoverBox>
-          )}
-          {this.props.isEditing && (
-            <Kb.Box onClick={this.props.onCancelEditing} style={styles.cancelEditing}>
-              <Kb.Text style={styles.cancelEditingText} type="BodySmallSemibold">
-                Cancel
-              </Kb.Text>
-            </Kb.Box>
-          )}
-          <input
-            type="file"
-            style={styles.hidden}
-            ref={this._filePickerSetRef}
-            onChange={this._pickFile}
-            multiple={true}
-          />
-          <Kb.Input
-            className={'mousetrap' /* className needed so key handler doesn't ignore hotkeys */}
-            autoFocus={false}
-            small={true}
-            style={styles.input}
-            ref={this._inputSetRef}
-            hintText={hintText}
-            hideUnderline={true}
-            onChangeText={this._onChangeText}
-            uncontrolled={true}
-            multiline={true}
-            rowsMin={1}
-            rowsMax={10}
-            onKeyDown={this._onKeyDown}
-            onEnterKeyDown={this._onEnterKeyDown}
-          />
-          {this.props.isExploding && !this.props.isEditing && !this.state.hasText && (
-            // This is the `boom!` icon in the placeholder: “Write an exploding message boom!”
-            <Kb.Icon
-              color={Styles.globalColors.black_20}
-              fontSize={34}
-              hoverColor={Styles.globalColors.black_20}
-              onClick={this._inputFocus}
-              style={Kb.iconCastPlatformStyles(styles.boomIcon)}
-              type="iconfont-boom"
-            />
-          )}
-          {this.props.showingMenu && (
-            <SetExplodingMessagePopup
-              attachTo={this.props.getAttachmentRef}
+      <KeyEventHandler
```

Comment:
didn't change anything here except wrap with KeyEventHandler and rename _globalKeyDownHandler